### ISSUE TITLE
Check that instances are Pixelfed ones

### ISSUE THREAD

**Issue by stom79** [H-PixelDroid/PixelDroid#175]
Currently, it's also possible to connect a Mastodon/Pleroma accounts with the app. Maybe it could be useful to use [nodeinfo](https://github.com/jhass/nodeinfo) before the oauth process.

endpoint: `/.well-known/nodeinfo`

Example:

https://pixelfed.social/.well-known/nodeinfo

=> https://pixelfed.social/api/nodeinfo/2.0.json

And check that the software name in the reply matches **pixelfed**

```
"software":{"name":"pixelfed","version":"0.10.9"}
```

**Comment by Wv5twkFEKh54vo4tta9yu7dHa3** [comment]
If someone wants to use the app with Mastodon or Pleroma, I'm not sure we should deny them that.

Maybe checking the /api/v1/instance endpoint and looking if the "version" string contains "Pixelfed" and then putting up a warning or something like that would be a good idea though.

**Comment by stom79** [comment]
Yes, I didn't mean to deny the access. A warning message is better.
IMO nodeinfo is the endpoint to use as it is built to give that information through several social softwares.

**Comment by Wv5twkFEKh54vo4tta9yu7dHa3** [comment]
I was thinking of using the instance endpoint since we request that already in the OAuth flow, and if we just want to check if it's Pixelfed or not it should be fine (other networks will just all be in the "not pixelfed" bucket anyway)

**Comment by stom79** [comment]
Yes, that avoids other calls. But I am not sure that in a long run this indication will be still there

`"version": "2.7.2 (compatible; Pixelfed 0.10.9)`

**Comment by Wv5twkFEKh54vo4tta9yu7dHa3** [comment]
Let's ping @dansup for that, I'm not sure either

**Comment by Wv5twkFEKh54vo4tta9yu7dHa3** [comment]
Since we'll have to request that anyway (for #185), we'll probably implement a warning using the endpoint as you suggested.
